TileDB: add support for building overviews on rasters stored as TileDB groups

rouault opened the pull request:
Supports the CREATE_GROUP=YES creation option to create TileDB raster datasets as a TileDB group, to make it cleaner to store overview within it.

See https://github.com/rouault/gdal/blob/tiledb_overview/frmts/tiledb/overview_model.rst for a description on how TileDB overviews are stored.

@normanb / @sgillies : this is mostly based on https://docs.google.com/document/d/1B-NizsoJn79W_RV3ujg6K1D5xJmWEja29Bv3323uYCU/edit#heading=h.p8z7kqfrb9st (with the removal of creating overviews on non-group dataset)

sgillies commented on frmts/tiledb/overview_model.rst:
```diff
@@ -0,0 +1,104 @@
+
+The group has the following TileDB metadata item:
+
+* ``dataset_type``: set to ``raster_overviews``
```
Maybe `dataset_type: raster_ovr_v1` with the same naming change as I proposed for `raster_pyr_v1`?

sgillies reviewed (commented):
@rouault @normanb I'm new to this work, made a quick scan this afternoon. I've got a few suggestions for dataset type names and group/array layout. Norman and I will also sync on this so that we're of one mind on it 😄

rouault commented on frmts/tiledb/overview_model.rst:
```diff
@@ -0,0 +1,104 @@
+as a TileDB array whose name is the base name of the group, suffixed with ``_0``
+
+The first overview level created is a TileDB array, placed within that group,
+and whose name is the base name of the group, suffixed with ``_1``. And so on.
```
I'm OK with that proposed change

rouault commented on frmts/tiledb/overview_model.rst:
```diff
@@ -0,0 +1,104 @@
+
+The group has the following TileDB metadata items:
+
+* ``dataset_type``: set to ``raster``
```
I'm fine with that change, and the below one too

rouault commented:
@sgillies @normanb Did you sync together regarding Sean's above suggestions?

sgillies commented:
@rouault sorry for the delay! We did discuss and here are the conclusions:

1. Overview arrays should be named `l_0`, `l_1`, etc. 
2. Dataset_type metadata should remain `raster`.
3. We do not desire a sidecar .ovr group and the TileDB platform will not support it, so it would be best to delete that format variant.
5. We'd like to see `CREATE_GROUP=YES` be the default. Groups will be favored by the TileDB platform. We understand that this means a group with a single array for users who don't request overviews.

rouault commented:
> We did discuss and here are the conclusions:

I've updated the pull request with those changes, starting at commit "Revert "TileDB raster: implement BuildOverviews() for base dataset not in group mode").
As the CREATE_GROUP creation option is common to the raster and vector sides of the driver, I found it to be perhaps simpler to change its default to YES for both (at least to be able to present a default value in the XML metadata of GDAL_DMD_CREATIONOPTIONLIST), so I also changed it for vectors. I could revert that change if needed.

rouault commented:
> As the CREATE_GROUP creation option is common to the raster and vector sides of the driver, I found it to be perhaps simpler to change its default to YES for both (at least to be able to present a default value in the XML metadata of GDAL_DMD_CREATIONOPTIONLIST), so I also changed it for vectors. I could revert that change if needed.

@sgillies Any opinion regarding that?

sgillies commented:
@rouault we would prefer to not change the default for vectors now.

rouault commented:
> @rouault we would prefer to not change the default for vectors now.

ok, change for vectors reverted

sgillies commented:
Thank you @rouault ! I think this looks great.
